src/app/services/myinfo/myinfo.util.ts

```diff
@@ -202,15 +203,18 @@ const compareSingleHash = (
 export const compareHashedValues = (
   responses: ProcessedFieldResponse[],
   hashes: IHashes,
-): Map<MyInfoAttribute, Promise<boolean>> => {
+): MyInfoComparePromises => {
   // Filter responses to only those fields with a corresponding hash
   const fieldsWithHashes = filterFieldsWithHashes(responses, hashes)
   // Map MyInfoAttribute to response
-  const myInfoResponsesMap = new Map<MyInfoAttribute, Promise<boolean>>()
+  const myInfoResponsesMap: MyInfoComparePromises = new Map()
   fieldsWithHashes.forEach((field) => {
     const attr = field.myInfo.attr
     // Already checked that hashes contains this attr
-    myInfoResponsesMap.set(attr, compareSingleHash(hashes[attr]!, field))
+    myInfoResponsesMap.set(
+      { fieldId: field._id, attr },
```

Comment:
This causes all responses to be unique, since a new object is created everytime for the key and and map does equality by reference.

See: https://stackoverflow.com/questions/61231359/javascript-map-with-composite-keys

Maybe we should stringify it? `${fieldId-attr}`

Reply:
yup, but isn't it okay because we don't expect there to be duplicate field IDs anyway? if we stringify then we strictly lose information, and have to parse the string to get back the field ID and attr later on.

Reply:
I see, ok. Maybe add a comment so it's clearer that all field-attr fields are unique and we are only using Maps for fast retrieval and not overwrites

Reply:
oh, I just had a better idea. we can make the key just the field ID, and change everything downstream to check for the field ID instead of the MyInfo attribute.

Reply:
more troublesome but probably worth it

Reply:
for sure